frontend/src/models/hand.js

```diff
@@ -41,7 +45,46 @@ export class Hand {
     return (
       this.cards.filter(card => card.rank === ranks.king).length < 5 &&
       this.cards.filter(card => card.value >= 10).length < 7 &&
-      this.cards[0].compareTo(jack.of(suits.diamonds)) < 0
+      this.highest().compareTo(jack.of(suits.diamonds)) < 0
+    );
+  }
+
+  fehlCardsBySuit(suit) {
+    return this.cards.filter(
+      card => card.suit === suit && card.isTrump() === false
     );
   }
+
+  mustServeSuit(suit) {
+    return this.fehlCardsBySuit(suit).length > 0;
+  }
+
+  trumps() {
+    return this.cards.filter(card => card.isTrump() === true);
+  }
+
+  hasTrumps() {
+    return this.trumps().length > 0;
+  }
+
+  getBlankAce(suit) {
+    let fehlCards = this.fehlCardsBySuit(suit);
+    return fehlCards.length === 1 && fehlCards[0].value === 11
+      ? fehlCards[0]
+      : undefined;
+  }
+
+  hasBlankAce(suit) {
+    return this.getBlankAce(suit) ? true : false;
+  }
+
+  getBlankAces() {
+    let aces = [];
+    [suits.clubs, suits.spades, suits.hearts].forEach(suit => {
```

Comment:
Very true, but right now solos aren't implemented, therefore I didn't implement something similar. I had this thought too, though 😀 If you implemented it on the go 👍🏻  Otherwise just add a todo an we'll have something to remember where we need to do some little refactoring.